hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/transactions/token/HapiTokenCreate.java

```diff
@@ -318,8 +325,16 @@ protected Consumer<TransactionBody.Builder> opBodyDef(final HapiSpec spec) throw
                                     k -> b.setAdminKey(spec.registry().getKey(k)));
                             freezeKey.ifPresent(
                                     k -> b.setFreezeKey(spec.registry().getKey(k)));
-                            supplyKey.ifPresent(
-                                    k -> b.setSupplyKey(spec.registry().getKey(k)));
+                            if (supplyKeyIsContractReference) {
```

Comment:
What this is doing isn't obvious from merely the test against `supplyKeyIsContractReference`.  Especially the `else`.  Comment?